packages/devtools/lib/src/timeline/frames_bar_chart.dart

```diff
@@ -59,6 +81,124 @@ class FramesBarChart extends CoreElement {
       _selectedFrameController.add(selectedFrame?.frame);
     }
   }
+*/
+}
+
+class PlotlyDivGraph extends CoreElement {
+  bool _processDatum;
+  Timer timer;
+  final FramesBarChart framesBarChart;
+  final TimelineFrame frame;
+  final Map<int, TimelineFrame> _frames = {};
+
+  ReadWriteMutex mutex = ReadWriteMutex();
+
+  List<int> dataIndexes = [];
+  List<num> cpuDurations = [];
+  List<num> gpuDurations = [];
+
+  FramesBarPlotly plotlyChart;
+
+  PlotlyDivGraph(this.framesBarChart, this.frame, [bool datum = true])
+      : this._processDatum = datum,
+        super('div') {
+    element.id = 'graph';
+    element.style
+      ..height = '100%'
+      ..width = '100%';
+  }
+
+  int frameIndex = 0;
+  bool _createdPlot = false;
+
+  // This routine should only be called using the mutex.acquireWrite() as it's
+  // destructive to our lists collecting index, and durations which are filled
+  // when onFrameAdded event and we chunk these to the our plotly chart to not
+  // cause the charting to lag.
```

Comment:
nit: consider re-writing this comment? Looks like there might be a typo ("we chunk these to the our") and in general I'm finding it hard to parse.  Perhaps split into two sentences.

Reply:
Done